R/utils.R

```diff
@@ -98,22 +98,40 @@ as_binary <- function(x) {
   as.integer(x)
 }
 
-random_string <- function(size = NULL) {
-  # creates a random string using tempfile() which does better to not affect the
-  # random seed
+#' random string function that does not alter the seed.
+#' 
+#' simple wrapper around `stringi::stri_rand_strings()``
+#' 
+#' @param n n
+#' @param length length
+#' @param pattern pattern
+#' @keywords internal
+#' @noRd
+random_string <- function(n = 1, length, pattern = "[A-Za-z0-9]") {
   # https://github.com/ycphs/openxlsx/issues/183
   # https://github.com/ycphs/openxlsx/pull/224
-  res <- basename(tempfile(""))
+  
+  if (missing(length)) length <- 16
 
-  if (is.null(size)) {
-    return(res)
-  }
+  rng <- get(".Random.seed", .GlobalEnv)
+
+  # try to get a previous openxlsx2 seed and use this as random seed
+  try(openxlsx2_seed <- get(".openxlsx2.seed", .GlobalEnv), TRUE)
 
-  size <- as.integer(size)
-  stopifnot(size >= 0L)
-  while (nchar(res) < size) {
-    res <- paste0(res, random_string())
+  if (exists("openxlsx2_seed")) {
+    # found one, change the global seed for stri_rand_strings
+    assign(".Random.seed", openxlsx2_seed, .GlobalEnv)
```

Comment:
`globalenv()` may be better to use

Reply:
Usually I would have used this, but when setting the seed I found this. Any specific reason?